src/Microsoft.ML.Onnx/TransformerChainOnnxConverter.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.UniversalModelFormat.Onnx;
+
+namespace Microsoft.ML.Model.Onnx
+{
+    public class TransformerChainOnnxConverter
+    {
+        public static ModelProto Convert<T>(TransformerChain<T> chain, IDataView inputData, HashSet<string> inputColumnNamesToDrop=null, HashSet<string> outputColumnNamesToDrop=null) where T : class, ITransformer
```

Comment:
>HashSet<string> inputColumnNamesToDrop=null, HashSet<string> outputColumnNamesToDrop [](start = 92, length = 84)

While I know why we have these from the *command's* perspective (since that kind of has to be a one-stop-shop) to do it from the *API*'s perspective is incorrect, since it complicates the call enormously. If someone wants to drop a column on `inputData`, they should just do that as they do literally everywhere else in the API... by dropping that column. Similarly, it seems that if someone does not want to include transform outputs produced, they ought to add something to the transform chain (again, dropping columns) to limit the output to only those columns they care about. At least, so I think.

Reply:
Sounds good. I will make those parameter not accessible to users. On the other hand, I am not sure if I can reproduce some old tests without the two argument, so having an internal version with the two arguments might be a backup plan.

---
In reply to: [245216200](https://github.com/dotnet/machinelearning/pull/2013#discussion_r245216200) [](ancestors = 245216200)